src/core/expr/fexpr_fillna.cc

```diff
@@ -139,7 +138,10 @@ class FExpr_FillNA : public FExpr_Func {
 
         for (size_t i = 0; i < wf.ncols(); ++i) {
           Column col_orig = Column(wf.get_column(i));
-          Column col_cond = make_isna_col(wf.retrieve_column(i));
+          Column col_cond = wf.retrieve_column(i);
+          bool is_void_column = col_cond.stype() == SType::VOID;
+          col_cond = is_void_column? Const_ColumnImpl::make_bool_column(col_cond.nrows(), true)
+                                   : Column(new Isna_ColumnImpl(std::move(col_cond)));
```

Comment:
What about creating a static method `Isna_ColumnImpl::make_isna_column(Column&& col)` that will perform the void check and have this shortcut to bool column?